structs.go

```diff
@@ -432,6 +432,12 @@ type GuildBan struct {
 	GuildID string `json:"guild_id"`
 }
 
+// A GuildBans stores data for a banned user.
+type GuildBans struct {
```

Comment:
I think we should change GuildBan to this structure.

We should add GuildID to the GuildBanAdd and GuildBanRemove structs, and change the embedded struct to User to match the discord api: 
https://discordapp.com/developers/docs/topics/gateway#guild-ban-add

So:
type GuildBanAdd struct {
  *User
  GuildID string
}

What do you think?

@bwmarrin?